cloudevents/conversion.py

```diff
@@ -0,0 +1,201 @@
+            )
+        if hasattr(raw_ce, "get"):
+            specversion = raw_ce.get("specversion", None)
+        else:
+            raise cloud_exceptions.MissingRequiredFields(
+                "Failed to read specversion from both headers and data. "
+                f"The following deserialized data has no 'get' method: {raw_ce}"
+            )
+
+    if specversion is None:
+        raise cloud_exceptions.MissingRequiredFields(
+            "Failed to find specversion in HTTP request"
+        )
+
+    event_handler = _obj_by_version.get(specversion, None)
+
+    if event_handler is None:
+        raise cloud_exceptions.InvalidRequiredFields(
+            f"Found invalid specversion {specversion}"
+        )
+
+    event = marshall.FromRequest(
+        event_handler(), headers, data, data_unmarshaller=data_unmarshaller
+    )
+    attrs = event.Properties()
+    attrs.pop("data", None)
+    attrs.pop("extensions", None)
+    attrs.update(**event.extensions)
+
+    if event.data == "" or event.data == b"":
+        # TODO: Check binary unmarshallers to debug why setting data to ""
+        # returns an event with data set to None, but structured will return ""
+        data = None
+    else:
+        data = event.data
+    return event_type.create(attrs, data)
+
+
+def _to_http(
+    event: AnyCloudEvent,
+    format: str = converters.TypeStructured,
+    data_marshaller: types.MarshallerType = None,
+) -> typing.Tuple[dict, typing.Union[bytes, str]]:
+    """
+    Returns a tuple of HTTP headers/body dicts representing this cloudevent
+
+    :param format: constant specifying an encoding format
+    :param data_marshaller: Callable function to cast event.data into
+        either a string or bytes
+    :returns: (http_headers: dict, http_body: bytes or str)
+    """
+    if data_marshaller is None:
+        data_marshaller = _marshaller_by_format[format]
+
+    if event["specversion"] not in _obj_by_version:
+        raise cloud_exceptions.InvalidRequiredFields(
+            f"Unsupported specversion: {event['specversion']}"
+        )
+
+    event_handler = _obj_by_version[event["specversion"]]()
+    for attribute_name in event:
+        event_handler.Set(attribute_name, event[attribute_name])
+    event_handler.data = event.data
+
+    return marshaller.NewDefaultHTTPMarshaller().ToRequest(
+        event_handler, format, data_marshaller=data_marshaller
+    )
+
+
+def to_structured(
+    event: AnyCloudEvent,
+    data_marshaller: types.MarshallerType = None,
+) -> typing.Tuple[dict, typing.Union[bytes, str]]:
+    """
+    Returns a tuple of HTTP headers/body dicts representing this cloudevent. If
+    event.data is a byte object, body will have a data_base64 field instead of
+    data.
+
+    :param event: CloudEvent to cast into http data
+    :param data_marshaller: Callable function to cast event.data into
+        either a string or bytes
+    :returns: (http_headers: dict, http_body: bytes or str)
+    """
```

Comment:
```suggestion
    """
    Returns a tuple of HTTP headers/body dicts representing this Cloud Event. 
    
    If event.data is a byte object, body will have a `data_base64` field instead of
    `data`.

    :param event: The event to be converted.
    :param data_marshaller: Callable function to cast event.data into
        either a string or bytes
    :returns: (http_headers: dict, http_body: bytes or str)
    """
```